concourse/client/util.py

```diff
@@ -13,20 +13,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
 import time
 import typing
+from contextlib import contextmanager
+from datetime import (
+    datetime,
+    timedelta,
+)
 
+from ci.util import (
+    info,
+    warning,
+)
 from whd.model import (
     PullRequestEvent,
 )
 from .model import (
     Job,
     PipelineConfigResource,
+    PipelineResource,
     ResourceType,
     ResourceVersion,
 )
 
 
+class PinningFailed(Exception):
```

Comment:
I think these are called `xyError` by convention (except the base Exception).
```suggestion
class PinningFailedError(Exception):
```